### PR TITLE
Refactor e2e test cypress cross browser

### PR DESCRIPTION
**Pull request by RajaBellebon**
## Description
* Bumping `cypress` to the latest version that enables cross browser tests
* Adding support for FF
* Splitting the test files to run them in Parallel
* General refactor

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -15,20 +15,18 @@ install_steps: &install_steps
     steps:
         - checkout
         - restore_cache:
-              name: Restore node_modules cache
```
Comment (RajaBellebon) [inline .circleci/config.yml]: Revert all changes to the config except the install for Cypress.

File: cypress.json [hunk 2]
```diff
@@ -1,5 +1,4 @@
 {
   "baseUrl": "http://localhost:8000/cypress-tests",
-  "screenshotOnHeadlessFailure": false,
-  "videoRecording": false
+  "video": false
```
Comment (RajaBellebon) [inline cypress.json]: check if we use true if we pay for it

File: package.json [hunk 3]
```diff
@@ -58,10 +59,10 @@
     "jest": "^24.8.0",
     "jest-emotion": "^10.0.11",
     "jest-in-case": "^1.0.2",
+    "lint-staged": "^8.1.5",
```
Comment (RajaBellebon) [inline package.json]: I need to remove it as per the discussion, it should come with HUSKY - TBC

File: packages/react-select/package.json [hunk 4]
```diff
@@ -22,7 +22,7 @@
     "react-transition-group": "^4.3.0"
   },
   "devDependencies": {
-    "cypress": "^1.4.1",
+    "cypress": "^4.1.0",
```
Comment (RajaBellebon) [inline packages/react-select/package.json]: Remove it from devDep

File: .circleci/config.yml (hunk 1 above)
Comment (RajaBellebon) [reply .circleci/config.yml]: done

File: cypress.json (hunk 2 above)
Comment (RajaBellebon) [reply cypress.json]: let's no record anything for now

File: package.json (hunk 3 above)
Comment (RajaBellebon) [reply package.json]: done but jfyi - getting that locally

```
 g add . && g commit -m "Review as per feedback"
Found '/Users/rbellebon/react-select/.nvmrc' with version <10>
Now using node v10.13.0 (npm v6.4.1)
Found '/Users/rbellebon/react-select/.nvmrc' with version <10>
Now using node v10.13.0 (npm v6.4.1)
husky > npm run -s precommit (node v10.13.0)

Found 0 errors
sh: lint-staged: command not found

husky > pre-commit hook failed (add --no-verify to bypass)```

File: packages/react-select/package.json (hunk 4 above)
Comment (RajaBellebon) [reply packages/react-select/package.json]: done

**Comment by flexdinesh** [comment]
Hey @RajaBellebon! Is this PR ready for review yet?

**Comment by RajaBellebon** [comment]
hey @flexdinesh - It was but with @gwyneplaine we discovered that we are not able to make the cypress run in CI due to some cache issues. If you know how to fix this that would be 🏆

**Comment by flexdinesh** [comment]
Thanks @RajaBellebon I'll try to take a look

**Comment by RajaBellebon** [comment]
@flexdinesh - hope you can find why we can't use version 4 in CircleCI :(

**Comment by RajaBellebon** [comment]
> Hi @RajaBellebon,
> 
> It appears Cypress was not being detected due to some kind of issue in the `yarn.lock` file.
> 
> Out of interest I re-created it and Cypress now works within CircleCI. Going forward with this `yarn.lock` change however, I think the issue needs to be narrowed now before we can merge this. From my knowledge we shouldn't be doing such a mass update of a `yarn.lock` file - but I am unsure about this and will need another contributor to comment.
> 
> Further, it now errors out when it runs one of the tests in the browser with your `--browser` option now specified, removing `--browser` locally caused the tests to run successfully.
> 
> See offending test and the comment above in the below screenshot:
> 
> hb8dkG
> 
> Could you explain your thinking behind adding in the `--browser` instead of just leaving it to run silently?
> 
> otN4BH
> 
> Thank you.

**Comment by RajaBellebon** [comment]
Sorry @bladey - thanks for looking into it!

Latest version of Cypress now, enables to run test in Chrome & Firefox - they recommend using the `--browser` flag.

I'll have a look and make everything working if possible 🤞 .

cc @flexdinesh

**Comment by bladey** [comment]
Thanks @RajaBellebon!

**Comment by flexdinesh** [comment]
@RajaBellebon I validated everything locally and ran into a small issue. It's a cypress thing and discussed here — https://github.com/cypress-io/cypress/issues/6813#issuecomment-661890295

As mentioned in the issue page, after upgrading cypress version to `^4.11.0` I was able to run the tests. If we make that change to the PR we can get it merged. 

cc @bladey

File: package.json [hunk 5]
```diff
@@ -47,7 +47,7 @@
     "coveralls": "^2.11.12",
     "cross-env": "^5.1.3",
     "css-loader": "^0.28.7",
-    "cypress": "^1.4.1",
+    "cypress": "^4.1.0",
```
Comment (flexdinesh) [inline package.json:50]: Can we upgrade cypress version to `^4.11.0`? More info here — https://github.com/JedWatson/react-select/pull/3957#issuecomment-667838702

Comment (RajaBellebon) [reply package.json:50]: yep let me do that now :)

**Pull request merged**